src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,103 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private static final double GROWTH_FACTOR = 1.5;
+    private T[] elements;
+    private int size;
+
+    public ArrayList() {
+        this.elements = (T[]) new Object[DEFAULT_CAPACITY];
```

Comment:
```suggestion
       elements = (T[]) new Object[DEFAULT_CAPACITY];
```
we don't need `this` [here](https://stackoverflow.com/questions/2411270/when-should-i-use-this-in-a-class/2411283#2411283)